### DISCUSSION TITLE
Pyload in Docker not saving download-directory

### DISCUSSION THREAD

**Discussion by mihawk90** [pyload/pyload#4382] [Help] [answered]
Hi there,

I was hoping to get some input on a small issue I'm having.
I am using [docker-pyload-ng](https://github.com/linuxserver/docker-pyload-ng/) on a TrueNAS install (and yes I know the Container isn't official, which is why I'm not filing a bug). It's all working well except for one oddity mentioned in the title: The Download directory I'm setting in the general settings isn't persisting and I don't really have an idea why.

I added a persistent volume for /config so things get saved to my actual storage pool.
When I go into the Pyload General settings and change my Downloads directory it saves it just fine. When I open a shell for the Docker container and check the settings file it is indeed saved in there:

```
folder storage_folder : "Download folder" = /mnt/downloads
```

However, after a restart it's back to the default `/downloads/`.

So I thought maybe I set up the Persistent Volume wrong, but that doesn't seem to be the case because it is saving other settings. I enabled "Skip login if single user" and that persists after a restart of the Container just fine:

```
bool autologin : "Skip login if single user" = True
```

Since both are in the same file, permissions are not the issue (especially since it does save it into the file, it just resets after the restart), so I'm wondering what it might be.

Are there any circumstances where Pyload resets it to default? I thought maybe when the path is non-existent. I used `/mnt/downloads/pyload` before and that doesn't exist, but `/mnt/downloads` does:
```
# ls /mnt -l
total 1
drwxr-xr-x 2 root root 2 Oct 21 16:47 downloads
```

Maybe someone has an idea? :thinking: 

PS: Yes I know I could just change the mountpoint to `/downloads`, but I'm curious why it's saving one setting but not the other.

**Comment by mihawk90** [comment]
oof... wow... seems I found it, of course after posting... was apparently looking at the wrong thing before:

https://github.com/linuxserver/docker-pyload-ng/blob/d7b359198b3912f66634770d211fcb385fc6b007/root/etc/s6-overlay/s6-rc.d/svc-pyload/run#L8

They pass the path when launching pyload.

However... I'm wondering whether that really should save to the config file? I feel like it should overwrite the Storage-Path at runtime but it shouldn't save to the config file. Ideally the WebUI could also show a warning that the field is not being used due to a commandline argument being passed. Maybe that's just me though :eyes:

**Comment by GammaC0de** [comment, marked as answer]
https://github.com/pyload/pyload/commit/b31b37e396f7772c64b269e040754c4f046fc547

**Reply by mihawk90** [reply]
Since this is just a log message, this still overwrites the config file, correct? Is this the way it's intended to work? I feel like a command-line switch is (most of the time) and on-demand non-permanent kind of thing, whereas a config file is by its nature permanent.
But maybe that's just differing philosophy :sweat_smile:

**Reply by GammaC0de** [reply]
Not saving the download directory in the config file can cause issues, for example https://github.com/pyload/pyload/blob/a65a968696f85c52c1f835d6d5dc22fd52fca12a/src/pyload/plugins/downloaders/MegaCoNz.py#L446

**Reply by mihawk90** [reply]
Ah right yeah, that makes sense. Would have to re-design more things then just the save to make it work :sweat_smile: Didn't think of that. Anyway, a warning in the log output is good enough for others running into this in the future I guess :)
Thank you!
